ra_aid/proc/interactive.py

```
@@ -1,90 +1,179 @@
 from typing import List, Tuple
 
-# Add macOS detection
-IS_MACOS = os.uname().sysname == "Darwin"
+import pyte
+from pyte.screens import HistoryScreen
 
+def render_line(line, columns: int) -> str:
+    """Render a single screen line from the pyte buffer (a mapping of column to Char)."""
+    return "".join(line[x].data for x in range(columns))
 
 def run_interactive_command(cmd: List[str]) -> Tuple[bytes, int]:
```

Ruff (0.8.2)

`run_interactive_command` is too complex (20 > 10)

(C901)